packages/link-bins/src/index.ts

```diff
@@ -122,7 +123,10 @@ async function getPackageBins (
   }
 
   if (R.isEmpty(pkg.bin)) {
-    throw new PnpmError('INVALID_PACKAGE_BIN', `Package in ${target} must have a non-empty bin field to get bin linked.`)
+    logger.warn({
```

Comment:
I was not right. In this case don't use `@pnpm/logger`. Use the warn fuction passed in with opts.

Reply:
that will also allow you to test it easily